src/parser/tests.rs

```diff
@@ -270,6 +274,39 @@ fn str_escape() {
     );
 }
 
+#[test]
+fn ascii_escape() {
+    assert_matches!(
+        parse("\"\\x[f\""),
+        Err(ParseError::InvalidEscapeSequence(..))
+    );
+    assert_matches!(
+        parse("\"\\x0\""),
+        Err(ParseError::InvalidEscapeSequence(..))
+    );
+    assert_matches!(
+        parse("\"\\x0z\""),
+        Err(ParseError::InvalidEscapeSequence(..))
+    );
+
+    assert_matches!(
+        parse("\"\\x80\""),
+        Err(ParseError::InvalidAsciiEscapeCode(..))
+    );
+    assert_matches!(
+        parse("\"\\xab\""),
+        Err(ParseError::InvalidAsciiEscapeCode(..))
+    );
+    assert_matches!(
+        parse("\"\\xFF\""),
+        Err(ParseError::InvalidAsciiEscapeCode(..))
+    );
```

Comment:
Might be worth to also add some tests ensuring that the escape codes are ignored in multi-line strings